filesystem/filesystem.go

```diff
@@ -59,6 +58,48 @@ func stat(s string) (*protocol.Dir, protocol.QID, error) {
 	return d, q, nil
 }
 
+func NewServer(opts ...ServerOpt) (*FileServer, error) {
+	s := &FileServer{
+		files: make(map[protocol.FID]*file),
+		trace: nologf,
+	}
+
+	for _, opt := range opts {
+		if err := opt(s); err != nil {
+			return nil, err
+		}
+	}
+
+	return s, nil
+}
+
+func Root(root string) ServerOpt {
+	return func(s *FileServer) error {
+		s.rootPath = root
+		return nil
+	}
+}
+
+func IOunit(size protocol.MaxSize) ServerOpt {
+	return func(s *FileServer) error {
+		s.IOunit = size
+		return nil
+	}
+}
+
+func Trace(tracer protocol.Tracer) ServerOpt {
+	return func(s *FileServer) error {
+		if tracer == nil {
+			return errors.New("tracer cannot be nil")
```

Comment:
or, hey, in here, 
tracer = func(string, ...interface{}) {}
for convenience, we can do that later.